server/src/main/webapp/WEB-INF/rails/webpack/views/pages/new-environments/spec/environments_widget_spec.tsx

```diff
@@ -49,67 +43,99 @@ describe("Environments Widget", () => {
   afterEach(helper.unmount.bind(helper));
 
   it("should render collapsible panel for each environment", () => {
+    const xmlEnv        = data.xml_environment_json();
+    const configRepoEnv = data.config_repo_environment_json();
+    const env           = data.environment_json();
     mountModal([xmlEnv, configRepoEnv, env]);
+
     expect(helper.byTestId("collapsible-panel-for-env-" + xmlEnv.name)).toBeInDOM();
     expect(helper.byTestId("collapsible-panel-for-env-" + configRepoEnv.name)).toBeInDOM();
     expect(helper.byTestId("collapsible-panel-for-env-" + env.name)).toBeInDOM();
   });
 
   it("should render environment header for all the environments", () => {
+    const xmlEnv        = data.xml_environment_json();
+    const configRepoEnv = data.config_repo_environment_json();
+    const env           = data.environment_json();
     mountModal([xmlEnv, configRepoEnv, env]);
+
     expect(helper.byTestId("environment-header-for-" + xmlEnv.name)).toBeInDOM();
     expect(helper.byTestId("environment-header-for-" + configRepoEnv.name)).toBeInDOM();
     expect(helper.byTestId("environment-header-for-" + env.name)).toBeInDOM();
   });
 
   it("should render environment body for all the environments", () => {
+    const xmlEnv        = data.xml_environment_json();
+    const configRepoEnv = data.config_repo_environment_json();
+    const env           = data.environment_json();
     mountModal([xmlEnv, configRepoEnv, env]);
+
     expect(helper.byTestId("environment-body-for-" + xmlEnv.name)).toBeInDOM();
     expect(helper.byTestId("environment-body-for-" + configRepoEnv.name)).toBeInDOM();
     expect(helper.byTestId("environment-body-for-" + env.name)).toBeInDOM();
   });
 
   it("should render delete icon for all the environments", () => {
-    mountModal([xmlEnv, configRepoEnv, env]);
+    mountModal([data.xml_environment_json(), data.config_repo_environment_json(), data.environment_json()]);
     const deleteIcons = helper.allByTestId("Delete-icon");
+
     expect(deleteIcons).toHaveLength(3);
   });
 
   it("should render disabled delete icon for the environments that user can not administer", () => {
+    const xmlEnv          = data.xml_environment_json();
     xmlEnv.can_administer = false;
+    mountModal([xmlEnv, data.xml_environment_json()]);
+
+    m.redraw.sync();
 
-    mountModal([xmlEnv, configRepoEnv, env]);
     const deleteIcons = helper.allByTestId("Delete-icon");
-    expect(deleteIcons).toHaveLength(3);
 
+    expect(deleteIcons).toHaveLength(2);
     expect(deleteIcons[0].title).toBe(`You are not authorized to delete the '${xmlEnv.name}' environment.`);
     expect(deleteIcons[0]).toBeDisabled();
     expect(deleteIcons[1].title).toBeFalsy();
     expect(deleteIcons[1]).not.toBeDisabled();
-    expect(deleteIcons[2].title).toBeFalsy();
-    expect(deleteIcons[2]).not.toBeDisabled();
+  });
+
+  it("should render disabled delete icon for the environments that user can not administer", () => {
```

Comment:
How about `should render disabled delete icon for the environments defined in config repo and user can not administer`
as it verifies the same
https://github.com/gocd/gocd/pull/7320/files#diff-16caf4c51b6acb7dd2c1cb0bcadf106aR109